spec/concurrent/timer_task_spec.rb

```diff
@@ -243,7 +243,7 @@ def trigger_observable(observable)
       end
 
       it 'notifies all observers on timeout' do
-        subject = TimerTask.new(execution: 0.1, timeout: 0.1) { sleep }
+        subject = TimerTask.new(execution: 500, timeout: 0.1, run_now: true) { sleep }
```

Comment:
This change makes the test invalid. The idea here is this:

1. Create a task that begins its execution `0.1` seconds after creation.
2. Tell the task to issue a timeout error if the task does not complete after `0.1` seconds.
    - The timeout error should be issued `0.2` seconds after `subject.execute` is called.
3. Add an observer to track the results.
4. Execute the task, beginning the `0.1` second timer before the block is run.
5. Wait `1.0` seconds then check for a timeout. 

Since the total time of the task should be approximately `0.2` seconds, we should see the timeout error before the countdown latch expires. If we change the `execution_interval` to `500` seconds then the block will not execute before the latch expires. We should not see a timeout error. The fact that the test passes indicates an error, but I believe the error is with the handling of `timeout_interval` and not `execution_interval`.

I'll look at this in more detail over the weekend.